extensions/warp-ipfs/src/store/message/community_task.rs

```diff
@@ -0,0 +1,3142 @@
+                    community_id: self.community_id,
+                    community_document: self.document.clone(),
+                    invite: invite_doc.clone(),
+                },
+            )
+            .await?;
+            if let Err(_e) = self.request_key(&did_key.clone()).await {}
+        }
+
+        Ok(CommunityInvite::from(invite_doc))
+    }
+    pub async fn delete_community_invite(&mut self, invite_id: Uuid) -> Result<(), Error> {
+        let own_did = &self.identity.did_key();
+
+        let mut is_targeting_self = false;
+        if let Some(invite) = self.document.invites.get(&invite_id.to_string()) {
+            if let Some(target) = &invite.target_user {
+                if target == own_did {
+                    is_targeting_self = true;
+                }
+            }
+        }
+        if !is_targeting_self
+            && !self
+                .document
+                .has_permission(own_did, &CommunityPermission::ManageInvites)
+        {
+            return Err(Error::Unauthorized);
+        }
+
+        self.document.invites.swap_remove(&invite_id.to_string());
+        self.set_document().await?;
+
+        let _ = self
+            .event_broadcast
+            .send(MessageEventKind::DeletedCommunityInvite {
+                community_id: self.community_id,
+                invite_id,
+            });
+
+        self.publish(
+            None,
+            CommunityMessagingEvents::UpdateCommunity {
+                community: self.document.clone(),
+                kind: CommunityUpdateKind::DeleteCommunityInvite { invite_id },
+            },
+            true,
+            vec![],
+        )
+        .await
+    }
+    pub async fn get_community_invite(
+        &mut self,
+        invite_id: Uuid,
+    ) -> Result<CommunityInvite, Error> {
+        match self.document.invites.get(&invite_id.to_string()) {
+            Some(invite_doc) => Ok(CommunityInvite::from(invite_doc.clone())),
+            None => Err(Error::CommunityInviteDoesntExist),
+        }
+    }
+    pub async fn accept_community_invite(&mut self, invite_id: Uuid) -> Result<(), Error> {
+        let own_did = &self.identity.did_key();
+        let invite_doc = self
+            .document
+            .invites
+            .get(&invite_id.to_string())
+            .ok_or(Error::CommunityInviteDoesntExist)?;
+
+        if let Some(target_user) = &invite_doc.target_user {
+            if own_did != target_user {
+                return Err(Error::CommunityInviteIncorrectUser);
+            }
+        }
+        if let Some(expiry) = &invite_doc.expiry {
+            if expiry < &Utc::now() {
+                return Err(Error::CommunityInviteExpired);
+            }
+        }
+
+        self.document.members.insert(own_did.clone());
+        if invite_doc.target_user.is_some() {
+            self.document
+                .invites
+                .swap_remove(&invite_doc.id.to_string());
+        }
```

Comment:
Should we also remove the invite from the document since it been accepted,  at least when it pertaining to a specific peer?

Reply:
Not sure what you mean, since that is exactly what is happening here.

Reply:
Sorry misread the code. Ignore :)